coding-concepts.js

```diff
@@ -21,17 +21,16 @@ const greeter = (name) => {
 }
 // console.log(greeter("LEARN Student"))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: Hello LEARN Student
+// b) Verify and explain: outout is undefined. this is because the function is missing a return statement which i missed when looking at the code
 
 // --------------------3) What will this log?
 
 const onlyOdds = [11, 12, 13, 14, 15].filter((number) => number % 2 !== 0)
 // console.log(onlyOdds)
 
-// a) Your answer:
-// b) Verify and explain:
-
+// a) Your answer: [11, 12, 13, 14, 15].filter((number) => number % 2 !== 0)
+// b) Verify and explain: the output is [ 11, 13, 15 ], I honestly dont understand why functions work the way they do. why does this work if the console.log is not invoking the parameter in next to the variable?? i thought it would just return what is says in onlyOdds.
```

Comment:
So we just need to break this down. `onlyOdds` is a variable that's holding data. The data is `[11, 12, 13, 14, 15].filter((number) => number % 2 !== 0)`.
This is just the HOF `.filter` being applied to the array `[11, 12, 13, 14, 15]`. So what does filter return? If we look at the syllabus, we can see that it returns a subset of the array it is being used on. The logic in the filter HOF is the condition that needs to be true in order for the value to be in the resulting array. Since filter returns an array of the values that meet the condition, we get `[ 11, 13, 15 ]`